docs/reference/mapping/runtime.asciidoc

```diff
@@ -0,0 +1,518 @@
+duration of the query without modifying the field itself.
+
+[discrete]
+[[runtime-use-cases]]
+=== Use cases
+Runtime fields are useful when working with log data
+(see <<runtime-examples,examples>>), especially when you're unsure about the
+data structure. Your search speed decreases, but your index size is much
+smaller and you can more quickly process logs without having to index them.
+
+Runtime fields are especially useful in the following contexts:
+
+* Adding fields to documents that are already indexed without having to reindex
+data
+* Immediately begin working on a new data stream without fully understanding
+the data it contains
+* Shadowing an indexed field with a runtime field to fix a mistake after
+indexing documents
+* Defining fields that are only relevant for a particular context (such as a
+visualization in {kib}) without influencing the underlying schema
+
+[discrete]
+[[runtime-compromises]]
+=== Compromises
+Runtime fields use less disk space and provide flexibility in how you access
+your data, but can impact search performance based on the computation defined in
+the runtime script.
+
+To balance search performance and flexibility, index fields that you'll
+commonly search for and filter on, such as a timestamp. {es} automatically uses
+these indexed fields first when running a query, resulting in a fast response
+time. You can then use runtime fields to limit the number of fields that {es}
+needs to calculate values for. Using indexed fields in tandem with runtime
+fields provides flexibility in the data that you index and how you define
+queries for other fields.
+
+Use the <<async-search,asynchronous search API>> to run searches that include
+runtime fields. This method of search helps to offset the performance impacts
+of computing values for runtime fields in each document containing that field.
+
+IMPORTANT: Queries against runtime fields are considered expensive. If
+<<query-dsl-allow-expensive-queries,`search.allow_expensive_queries`>> is set
+to `false`, expensive queries are not allowed and {es} will reject any queries
+against runtime fields.
+
+[discrete]
+[[runtime-mapping-fields]]
+=== Mapping a runtime field
+You map runtime fields by adding a `"runtime"` section under the mapping
+definition. Within that section, you
+<<modules-scripting-using,define a Painless script>>, which has access to the
+original `_source` and `doc_values` (the mapped field plus its values). At
+search time, the script runs and generates values for each scripted field.
+
+NOTE: You can define a runtime field in the mapping definition without a
+script. {es} will look in `_source` for a field with the same name as the
+runtime field and use values from that field at query time.
+
+Runtime fields are similar to the <<script-fields,`script_fields`>> parameter
+of the `_search` request, but also make the script results available for
+queries and aggregations.
```

Comment:
not for queries and aggregations, also sorting. Should rather say anywhere in a search request?

Reply:
I'll change to, "...make the script results available anywhere in a search request."